types/marked/index.d.ts

```diff
@@ -77,6 +78,7 @@ export namespace marked {
      * @param callback Function called when the markdownString has been fully parsed when using async highlighting
      * @return String of compiled HTML
      */
+    function parse(src: string, options: MarkedOptions & {async: true}, callback?: (error: any, parseResult: string) => void): Promise<string>;
```

Comment:
The return type is always `void` if a callback exists. We should make this different overrides like the `marked` function

Reply:
Got it!